### PR TITLE
Added non-mandatory kms key setup with versioning and mfa features

### PR DESCRIPTION
**Pull request by AwkwardBen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,4 +20,7 @@ Module usage:
 | name | The allocated to the s3 bucket in aws | - | yes |
 | acl | Amazon control list of access rights to the created bucket | - | yes |
 | environment | The environment the S3 is running in i.e. dev, prod etc | - | yes |
+| kms_alias | The alias name used for the kms key | - | no |
+| versioning_enabled | If the versioning is enabled for the created s3 bucket | true | no |
+| mfa_delete_enabled | If mfa is need for delete for s3 bucket | false | no |
```
Comment (marcinc) [inline README.md]: If you could reword that a little bit that would be good.

**Review by oyelekci — APPROVED** [review]
LGTM

**Comment by jaykeshur** [comment]
A few things:
- The s3 policy doesn't include being able to use the kms-id
- The s3 policy should include server side encryption using that key only.
- you need to create a user to do all of this

**Review by jaykeshur — CHANGES_REQUESTED** [review]
notes added

File: main.tf [hunk 2]
```diff
@@ -1,36 +1,165 @@
 
 data "aws_caller_identity" "current" { }
 
+resource "aws_kms_key" "s3_bucket_kms_key" {
+
+  count = "${var.kms_alias == "" ? 0 : 1}"
+
+  description = "A kms key for encrypting/decryting S3 bucket ${var.name}"
+  policy =
```
Comment (jaykeshur) [inline main.tf]: should this be empty?

**Review by jaykeshur — APPROVED** [review]

**Pull request merged**